Alignment/include/ActsAlignment/Kernel/Alignment.hpp

```diff
@@ -230,4 +230,65 @@ struct Alignment {
 };
 }  // namespace ActsAlignment
 
-#include "ActsAlignment/Kernel/Alignment.ipp"
+
+//Struct that takes care of grouping misaligned detector elements
+struct GroupedMisalignment {
+    std::vector<std::string> geoIDPrefixes;
+    Acts::Transform3 combinedMisalignmentDelta;
+    Acts::Vector3 centerOfGravity;
+    std::vector<Acts::Transform3> relativeTransforms;
+};
+
+//  function for creating grouped misalignment
+GroupedMisalignment createGroupedMisalignment(
+    const std::vector<Acts::DetectorElementBase*>& allElements,
+    const std::vector<std::string>& geoIDPrefixes);
+ // 
+Acts::Result<void> ActsAlignment::updateAlignmentParameters(
+    const Acts::GeometryContext& gctx,
+    const std::vector<Acts::DetectorElementBase*>& alignedDetElements,
+    const AlignedTransformUpdater& alignedTransformUpdater,
+    AlignmentResult& alignResult,
+    const GeometryHierarchyMap& geometryMap,
+    const AlignmentParameterSet& alignmentParameterSet) const {
+    
+    // defining the group pattern
+   std::set<Acts::GeometryContext::GeoID> group1Pattern = {
```

Comment:
These are examples for testing? If so, it might make sense to put this into a dedicated unit test that checks what happens with these inputs.

Also: these sets are explicit, right? 

It's not clear to me that this selects surface the way we discussed. Maybe @gagnonlg can comment if I'm missing the point.